docs/scripts/gen-faq.js

```diff
@@ -0,0 +1,174 @@
+// read.js
+const fs = require('fs')
+const yaml = require('js-yaml')
+const { Remarkable } = require('remarkable')
+const yamlPath = './docs/faq.yaml'
+
+try {
+  // Generating FAQ.mdx
+
+  if (!fs.existsSync(yamlPath)) {
+    //file exists
+    console.warn('.yaml File does not exists, skipping generating FAQ')
+    return 0
+  }
+
+  let fayYaml = fs.readFileSync('yamlPath', 'utf8')
+  let faq = yaml.load(fayYaml)
+
+  const tags = Array.from(
+    new Set(
+      faq
+        .map((el) => {
+          return el.tags
+        })
+        .flat(1)
+    )
+  )
+
+  data = `---
+id: faq
+title: Faq
+---
+
+
+
+import {Question, Faq} from '@theme/Faq'
+
+<Faq tags="${tags.join(' ')}"/>
+<br/><br/>
+
+`
+  md = new Remarkable()
+  faq.forEach((el) => {
+    react_tags = el.tags.map((tag) => {
+      return tag + '_src-theme-'
+    })
+    data += `<Question tags="question_src-theme- ${react_tags.join(' ')}">\n`
+    data += `${el.tags
+      .map((tag) => {
+        return '#' + tag
+      })
+      .join(' ')} <br/>\n`
+    data += md.render(`**Q**: ${el.q}`)
```

Comment:
It might make sense to have data as an string array and then do `data.push()` and in the end `data.join()` which would prevent forgetting to add `\n` at the end of statements.

Reply:
Will adress it with @k9ert as well tomorrow.